### PR TITLE
Updated Readme - added note about maven compile target failure

### PR DESCRIPTION
**Pull request by rkolesnev**
Updated Readme - added note about maven compile target failure due to a test-jar dependency resolution bug.

### CODE REVIEW

**Comment by astubbs** [comment]
Please rebase

File: src/docs/README.adoc [hunk 1]
```diff
@@ -692,6 +692,11 @@ git bisect start good bad
 git bisect run ./bisect.sh
 ----
 
+Note::
```
Comment (astubbs) [inline src/docs/README.adoc:695]: did you make a GH issue for this to be fixed as well? If not please do so. And link to the GH issue from the note in the readme.

**Review by astubbs — CHANGES_REQUESTED** [review]

File: src/docs/README.adoc [hunk 2]
```diff
@@ -692,6 +692,12 @@ git bisect start good bad
...
+mvn compile - Due to a bug in maven handling of test-jar dependencies - running `mvn compile` fails,
+unless `mvn install` or `mvn deploy` was executed previously to create test-jar of parallel-consumer-core.
+See https://github.com/confluentinc/parallel-consumer/issues/162
```
Comment (astubbs) [inline src/docs/README.adoc:698]:
```suggestion
See https://github.com/confluentinc/parallel-consumer/issues/162[issue #162]
```

File: src/docs/README.adoc [hunk 3]
```diff
@@ -692,6 +692,12 @@ git bisect start good bad
...
+and https://stackoverflow.com/questions/4786881/why-is-test-jar-dependency-required-for-mvn-compile
```
Comment (astubbs) [inline src/docs/README.adoc:699]:
```suggestion
and this https://stackoverflow.com/questions/4786881/why-is-test-jar-dependency-required-for-mvn-compile[Stack Overflow question].
```

File: src/docs/README.adoc [hunk 4]
```diff
@@ -692,6 +692,12 @@ git bisect start good bad
 git bisect run ./bisect.sh
 ----
 
+Note::
+mvn compile - Due to a bug in maven handling of test-jar dependencies - running `mvn compile` fails,
```
Comment (astubbs) [inline src/docs/README.adoc:696]:
```suggestion
`mvn compile` - Due to a bug in Maven's handling of test-jar dependencies - running `mvn compile` fails, Use `mvn test-compile` instead.
```

**Review by astubbs — CHANGES_REQUESTED** [review]

File: src/docs/README.adoc (hunk 4 above)
Comment (rkolesnev) [reply src/docs/README.adoc]: updated

File: src/docs/README.adoc (hunk 3 above)
Comment (rkolesnev) [reply src/docs/README.adoc]: updated.

File: src/docs/README.adoc (hunk 2 above)
Comment (rkolesnev) [reply src/docs/README.adoc]: updated.

**Review by astubbs — APPROVED** [review]

**Pull request merged**